### PR TITLE
FIX: array arguments not parsed correctly

### PR DESCRIPTION
**Pull request by SamSaffron**
- FIX: array arguments not parsed correctly
- Side quest ... don't use update! it calls validations and will now fail due to email validation

### CODE REVIEW

**Comment by SamSaffron** [comment]
side quest used to work, something changed in core ...

File: lib/ai_bot/personas/persona.rb [hunk 1]
```diff
@@ -122,16 +122,24 @@ def find_tool(partial)
...
+                begin
+                  JSON.parse(value)
+                rescue JSON::ParserError
+                  nil
+                end
+            end
+
+            if value
+              arguments[name.to_sym] = value if value
+            end
```
Comment (tgxworld) [inline lib/ai_bot/personas/persona.rb:141]: There is a double `if value` conditional here. Not sure why Github doesn't allow me to use code suggestion here.

Comment (tgxworld) [inline lib/ai_bot/personas/persona.rb:141]:
```
            if value
              if param[:type] == "array"
                value =
                begin
                  JSON.parse(value)
                rescue JSON::ParserError
                  nil
                end
              end

              arguments[name.to_sym] = value
            end
```

Comment (SamSaffron) [reply lib/ai_bot/personas/persona.rb:141]: Oops will fix

**Review by martin-brennan — APPROVED** [review]

**Pull request merged**